lms/static/js/student_account/register.js

```diff
@@ -0,0 +1,111 @@
+ */
+
+(function($, gettext) {
+    'use strict';
+    var buildModal = function(link, modal_class, content_class, close_button_class) {
+        // create the modal container with a title, close button and content area
+        var close_label = gettext('Close'),
+            title_id = 'modal-header-text',
+            modal = $('<div>', {
+                'class': modal_class
+            }),
+            content = $('<div>', {
+                'class': content_class,
+                role: 'dialog',
+                'aria-modal': true,
+                'aria-labelledby': title_id
+            }),
+            close_button = $('<button>', {
+                'aria-label': close_label,
+                'class': close_button_class
+            }),
+            title = $('<h2>', {
+                id: title_id
+            });
+        close_button.text(close_label);
+        content.append(close_button);
+        title.text($(link).text());
+        content.append(title);
+        modal.append(content);
+        return modal;
+    };
+    var buildIframe = function(link, content_selector) {
+        // create an iframe with contents from the link and set its height to match the content area
+        return $('<iframe>', {
+            src: link.href,
+            load: function() {
+                var iframe_head = $(this).contents().find('head');
+                // overwrite styles in child page
+                // hide top navigation and footer
+                var style = $('<style>', {'type': 'text/css'});
+                style.html(`
+                    .nav-skip, header#global-navigation, .wrapper-footer {
+                        display:none;
+                    }
+                    .container.about {
+                        padding: 0;
+                        min-width: auto;
+                    }
+                `);
+                iframe_head.append(style);
+                // set the iframe's height to fill the available space
+                $(this).css({
+                    height: $(content_selector).height()
+                });
+            }
+        });
+    };
+    var showModal = function(modal_selector) {
+        $(modal_selector).show();
+        // prevent scrolling of the background
+        $('body').addClass('open-modal');
+    };
+    var hideModal = function(modal_selector) {
+        $(modal_selector).hide();
+        $('body').removeClass('open-modal');
+        $(modal_selector).find('iframe').remove();
+    };
+    $(document).ready(function() {
+        var tos_link_selector = '.checkbox-honor_code .supplemental-link a',
+            close_button_class = 'modal-close-button',
+            close_button_selector = '.' + close_button_class,
+            content_class = 'modal-content',
+            content_selector = '.' + content_class,
+            modal_class = 'modal',
+            modal_selector = '.' + modal_class;
+        $('body').on('click', tos_link_selector, function(event) {
```

Comment:
@replaceafill Please add some vertical whitespace to this part of the code as well.